File: src/cleansia_android/core/src/main/java/cz/cleansia/core/notifications/DeviceRegistrationClient.kt
package cz.cleansia.core.notifications

/**
 * Per-app binding seam for the push-notification device endpoints. Each app
 * implements this over its own OpenAPI-generated `DeviceApi` client; the
 * shared [PushTokenRepository] depends only on this interface so the lifecycle
 * logic lives once in `:core`.
 *
 * Implementations return `true` only when the backend accepted the call so the
 * repository caches the last-registered token exclusively on success.
 */
interface DeviceRegistrationClient {

    suspend fun register(request: RegisterDeviceRequest): Boolean

    suspend fun unregister(deviceId: String): Boolean
}

File: src/cleansia_ios/CleansiaCore/Sources/CleansiaCore/Push/DeviceRegistrationClient.swift
import Foundation

public struct RegisterDeviceRequest: Equatable, Sendable {
    public let deviceId: String
    public let deviceToken: String
    public let platform: String

    public init(deviceId: String, deviceToken: String, platform: String) {
        self.deviceId = deviceId
        self.deviceToken = deviceToken
        self.platform = platform
    }
}

public protocol DeviceRegistrationClient: Sendable {
    func register(_ request: RegisterDeviceRequest) async -> ApiResult<Void>
    func unregister(deviceId: String) async -> ApiResult<Void>
}
